### DISCUSSION TITLE
How can i remove kustomization/flux-system

### DISCUSSION THREAD

**Discussion by wibed** [fluxcd/flux2#2869] [General] [answered]
i trying to remove flux without deleting the whole cluster.

and stuck at the following step: `kubectl -n flux-system delete kustomization flux-system`
it does not remove the resource, i have already removed any resource related to flux from the cluster.
i looked for any other related resource but cant seem to find it. 

i tried, but did not help:
  - deleting the crd's, but this just renders the kustomization not queriable.
  - `flux delete kustomization flux-system`

**Comment by souleb** [comment, marked as answer]
Flux sets a finalizer on the resource. Edit it with kubectl and delete the finalizer.

**Comment by wibed** [comment]
@souleb 

i tried patching the kustomization with

```sh
kubectl -n flux-system patch kustomization flux-system -p '{"metadata":{"finalizers":null}}'
```

it is the same approach id use on crd's. but it results in a
`Unsupported Media Type`- Error

**Reply by wibed** [reply]
i succeeded adding the `--type=merge`; to the argument list

**Comment by stefanprodan** [comment]
Please use `flux uninstall`, docs here: https://fluxcd.io/docs/installation/#uninstall

**Reply by AzySir** [reply]
> Please use `flux uninstall`, docs here: https://fluxcd.io/docs/installation/#uninstall

That would be great if it actually worked.. spending hours here trying to force delete the CRDs...

**Reply by stefanprodan** [reply]
You're probably using a Flux CLI version that doesn't match the in-cluster version.
